hazelcast/src/main/java/com/hazelcast/util/PhoneHome.java

```diff
@@ -204,6 +218,164 @@ private void fetchWebService(String urlStr) {
+        parameterCreator.addParam("cpy", Integer.toString(clusterClientStats.get(ClientType.PYTHON)));
+    }
+
+    private void addManCenterInfo(ExecutorService executor, Future<MCResponse> response,
+                                  int clusterSize, PhoneHomeParameterCreator parameterCreator) {
+        MCResponse mcResponse = null;
+        try {
+            mcResponse = response.get();
+        } catch (InterruptedException ignored) {
+            EmptyStatement.ignore(ignored);
+        } catch (ExecutionException ignored) {
+            EmptyStatement.ignore(ignored);
+        } finally {
+            executor.shutdown();
+        }
+
+        MCPhoneHomeInfo mcPhoneHomeInfo = getManCenterPhoneHomeInfo(mcResponse);
+        if (mcPhoneHomeInfo != null) {
+            String version = mcPhoneHomeInfo.getVersion();
+            Object license = mcPhoneHomeInfo.getLicense();
+            int responseCode = mcPhoneHomeInfo.getResponseCode();
+            if (responseCode == HttpURLConnection.HTTP_OK) {
+                if (license == null) {
+                    checkClusterSizeAndSetLicense(clusterSize, parameterCreator);
+                } else {
+                    parameterCreator.addParam("mclicense", license.toString());
+                }
+                parameterCreator.addParam("mcver", version);
+            } else {
+                parameterCreator.addParam("mclicense", "MC_CONN_ERR_" + responseCode);
+                parameterCreator.addParam("mcver", "MC_CONN_ERR_" + responseCode);
+            }
+        } else {
+            parameterCreator.addParam("mclicense", "MC_NOT_AVAILABLE");
+            parameterCreator.addParam("mcver", "MC_NOT_AVAILABLE");
+        }
+    }
+
+    private void checkClusterSizeAndSetLicense(int clusterSize, PhoneHomeParameterCreator parameterCreator) {
+        if (clusterSize <= 2) {
+            parameterCreator.addParam("mclicense", "MC_LICENSE_NOT_REQUIRED");
+        } else {
+            parameterCreator.addParam("mclicense", "MC_LICENSE_REQUIRED_BUT_NOT_SET");
+        }
+    }
+
+    private MCPhoneHomeInfo getManCenterPhoneHomeInfo(MCResponse mcResponse) {
+        InputStream inputStream = null;
+        InputStreamReader reader = null;
+        MCPhoneHomeInfo mcPhoneHomeInfo = null;
+        if (mcResponse == null) {
+            return mcPhoneHomeInfo;
+        }
+        try {
+            inputStream = mcResponse.getInputStream();
+            reader = new InputStreamReader(inputStream, "UTF-8");
+            JsonObject mcPhoneHomeInfoJson = JsonValue.readFrom(reader).asObject();
+            final String version = JsonUtil.getString(mcPhoneHomeInfoJson, "mcVersion");
+            final String license = JsonUtil.getString(mcPhoneHomeInfoJson, "mcLicense", null);
+            final int responseCode = mcResponse.getResponseCode();
+            mcPhoneHomeInfo = new MCPhoneHomeInfo(version, license, responseCode);
+        } catch (IOException ignored) {
```

Comment:
maybe adding some log here otherwise we might never notice if there is a bug in parsing part.